src/transformers/modeling_utils.py

```diff
@@ -2534,11 +2568,25 @@ def _fix_key(key):
                 if key.startswith(prefix):
                     key = ".".join(key.split(".")[1:])
                 param = model_state_dict[key]
+
+                # upcast in fp32 if any
+                target_dtype = dtype
+                if keep_in_fp32_modules is not None and any(
+                    module_to_keep_in_fp32 in key for module_to_keep_in_fp32 in keep_in_fp32_modules
```

Comment:
should also add a test of `dtype` being `float16` here.

Reply:
added in 8014c34